.github/workflows/go.yml

```
@@ -382,9 +382,60 @@ jobs:
       - name: test contract
         run: make test-contracts
 
-  # Tests the helloworld example, this indirectly ensures that the autonity
-  # docker image can be built since the helloworld example uses it. We don't
-  # want to use any cache here since we want a clean checkout before any code
+  docker_e2e_test:
+    needs: bootstrap-checkout
+    runs-on: ubuntu-latest
+    if: github.ref == 'refs/heads/develop'
+    steps:
+      - name: Set up Go
+        uses: actions/setup-go@v1
+        with:
+          go-version: 1.13
+        id: go
+
+      - name: Set GOPATH
+        # see https://github.com/actions/setup-go/issues/14
+        run: |
+          echo "##[set-env name=GOPATH;]$(dirname $GITHUB_WORKSPACE)"
+          echo "##[add-path]$(dirname $GITHUB_WORKSPACE)/bin"
+        shell: bash
+
+      - name: Cache code
+        uses: actions/cache@v1
+        env:
+          cache-name: cache-autonity-code
+        with:
+          path: /home/runner/work/autonity/autonity/
+          key: ${{ runner.os }}-aut-${{ env.cache-name }}-${{ github.sha }}
+          restore-keys: ${{ runner.os }}-aut-${{ env.cache-name }}-${{ github.sha }}
```

We shouldn't use restore keys  #504 , and this restore key makes no sense since it is the same as the main key.
```suggestion
          restore-keys: ${{ runner.os }}-aut-${{ env.cache-name }}-${{ github.sha }}
```